pyCalculations/fieldtracer.py

```diff
@@ -273,6 +219,129 @@ def static_field_tracer_3d( vlsvReader, coord_list, max_iterations, dx, directio
+                                  grid_variable = 'vg_b_vol'
+      keyword stop_condition: Boolean array (seed_coords.shape[0],)
+                              Determine when the iteration stop, for the vg trace only
+                              If not specified, it will always be True for each seed points.
+                              eg. def my_stop(points):
+                                    distances = np.linalg.norm(points[:,:],axis = 1)
+                                    return (distances <= lower_bound) | (distances >= upper_bound)
+      :returns:               fg:   points_traced --- Traced coordinates (a list of lists of 3-element coordinate arrays)
+                                 ex. points_traced[2][5][1]: at 3rd tracing step [2], the 6th point [5], y-coordinate [1]
+                                    note: Can convert output to a 3D numpy array if desired, with np.array(points_traced)
+                              vg:   points_traced --- a 3d numpy array [len(seed_coords)]
+      EXAMPLE:            vlsvobj = pytools.vlsvfile.VlsvReader(vlsvfile) 
+                          fg_b = vlsvobj.read_variable('fg_b')
+                          traces = static_field_tracer_3d( vlsvobj, [[5e7,0,0], [0,0,5e7]], 10, 1e5, direction='+', fg = fg_b )
+   '''
+
+   # Standardize input: (N,3) np.array
+   if type(seed_coords) != np.ndarray:
+      raise TypeError("Please give a numpy array.")
+
+   # Cache and read variables:
+   vg = None
+   fg = None
+
+   if isinstance(grid_var, str):
+      parts = grid_var.split("/")
+      for part in parts:
+         if part.startswith("fg"):
+            fg = grid_var
+            break
+         elif part.startswith("vg"):   
+            vg = grid_var   
+            # neighbors_vg = vlsvReader.read_variable_to_cache("vg_regular_interp_neighbors")
+            vg_cache = vlsvReader.read_variable_to_cache(vg)
+            break
+      else:
+         raise ValueError("Please give a valid string (eg. 'vg_b_vol')")
+   else:
+      raise TypeError("Please give a string")
+      # #   fg is already an ndarray
+      # if not isinstance(grid_var, np.ndarray):
+      #    raise TypeError("Keyword parameter fg does not seem to be a numpy ndarray.")
+      # elif fg.ndim!=4 or fg.shape[-1]!=3:
+      #    raise ValueError("Checking array supplied in fg keyword: fg[-1]={} (expected: 3), fg.ndim={} (expected: 4)".format(fg[-1], fg.ndim))
+         
+   # Recursion (trace in both directions and concatenate the results)
+   if direction == '+-':
+      backward = static_field_tracer_3d(vlsvReader, coord_list, max_iterations, dx, direction='-', grid_var = grid_var)
+      backward.reverse()
+      forward = static_field_tracer_3d(vlsvReader, coord_list, max_iterations, dx, direction='+', grid_var = grid_var)
+      return np.concatenate((backward,forward[:, 1:, :]), axis = 1)
+
+   multiplier = -1 if direction == '-' else 1   
```

Comment:
I don't think this is passed to the tracing subroutines now.

Reply:
Yes, and neither does dx